resqpy/property/property_common.py

```diff
@@ -307,111 +367,129 @@ def guess_uom(property_kind, minimum, maximum, support, facet_type = None, facet
-        z_units = rqet.find_tag(crs_node, 'VerticalUom').text.lower()
-        if xy_units == 'm' and z_units == 'm':
-            return 'm'
-        if xy_units == 'ft' and z_units == 'ft':
-            return 'ft'
-        return None
 
+def _guess_uom_get_crs_info(support):
     if support is None or not hasattr(support, 'extract_crs_root'):
         crs_node = None
     else:
         crs_node = support.extract_crs_root()
-    from_crs = crs_m_or_ft(crs_node)
+    return crs_node, _crs_m_or_ft(crs_node)
 
-    if property_kind in ['rock volume', 'pore volume', 'volume']:
-        if from_crs is None:
-            return None
-        if from_crs == 'ft' and property_kind == 'pore volume':
-            return 'bbl'  # seems to be Nexus 'ENGLISH' uom for pv out
-        return from_crs + '3'  # ie. m3 or ft3
-    if property_kind == 'depth':
-        if crs_node is None:
-            return None
-        return rqet.find_tag(crs_node, 'VerticalUom').text.lower()
-    if property_kind == 'cell length':  # todo: pass in direction facet to pick up xy_units or z_units
-        return from_crs
-    if property_kind in ['net to gross ratio', 'porosity', 'saturation']:
-        if maximum is not None and str(maximum) != 'unknown':
-            max_real = float(maximum)
-            if max_real > 1.0 and max_real <= 100.0:
-                return '%'
-            if max_real < 0.0 or max_real > 1.0:
-                return None
+
+def _guess_uom_depth(crs_node):
+    if crs_node is None:
+        return None
+    return rqet.find_tag(crs_node, 'VerticalUom').text.lower()
+
+
+def _guess_uom_volume(property_kind, from_crs, facet_type, facet):
+    if property_kind == 'fluid volume':
         if from_crs == 'm':
-            return 'm3/m3'
+            return 'm3'
         if from_crs == 'ft':
-            return 'ft3/ft3'
-        return 'Euc'
-    if property_kind == 'permeability rock' or property_kind == 'rock permeability':
-        return 'mD'
-    if property_kind == 'permeability thickness':
+            if facet_type == 'what' and facet == 'gas':
+                return '1000 ft3'  # todo: check units output by nexus for GIP
+            else:
+                return 'bbl'  # todo: check whether nexus uses 10^3 or 10^6 units
+        return None
+    if from_crs is None:
+        return None
+    if from_crs == 'ft' and property_kind == 'pore volume':
+        return 'bbl'  # seems to be Nexus 'ENGLISH' uom for pv out
+    return from_crs + '3'  # ie. m3 or ft3
+
+
+def _guess_uom_permeability(property_kind, crs_node):
+    if 'thickness' in property_kind:
         z_units = rqet.find_tag(crs_node, 'VerticalUom').text.lower()
         if z_units == 'm':
             return 'mD.m'
         if z_units == 'ft':
             return 'mD.ft'
         return None
-    if property_kind == 'permeability length':
+    else:
         xy_units = rqet.find_tag(crs_node, 'ProjectedUom').text.lower()
```

Comment:
This should use the z units instead of the xy units for the length component, I believe.

Reply:
Updated to return None if the xy / z units are mismatched, or the xyz units if matching.